### PR TITLE
Upgraded client-go to 1.14.0

### PR DESCRIPTION
**Pull request by yuchaoran2011**
The lifecycle test complements the controller unit tests, but from an integration test level.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -75,19 +75,19 @@ required = [
 
 [[constraint]]
   name = "k8s.io/api"
-  version = "kubernetes-1.13.4"
+  version = "kubernetes-1.14.0"
```
Comment (liyinan926) [inline Gopkg.toml]: Do we really need to upgrade to 1.14 given that it's just released?

Comment (yuchaoran2011) [reply Gopkg.toml]: I guess why not? But if you have any concerns, I'm fine waiting a bit (maybe after 1.14.1 is out).

Comment (liyinan926) [reply Gopkg.toml]: Yes, let's wait for 1.14 to bake a bit more.

Comment (yuchaoran2011) [reply Gopkg.toml]: Ok I'll make another PR containing just the added tests. Then we can merge this one later.

**Comment by yuchaoran2011** [comment]
K8s 1.14.2 is out but haven't had time to work on it. Closing it for now. Will re-open when I get to it.

**Pull request closed without merging**